browser/ui/webui/webcompat_reporter/webcompat_reporter_ui.cc

```diff
@@ -0,0 +1,178 @@
+#include "brave/components/brave_vpn/common/buildflags/buildflags.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/components/webcompat_reporter/browser/fields.h"
+#include "brave/components/webcompat_reporter/browser/webcompat_report_uploader.h"
+#include "brave/components/webcompat_reporter/resources/grit/webcompat_reporter_generated_map.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/grit/brave_components_resources.h"
+#include "components/language/core/browser/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "content/public/browser/web_ui_message_handler.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "ui/web_dialogs/web_dialog_delegate.h"
+#include "url/gurl.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/browser/brave_vpn/brave_vpn_service_factory.h"
+#include "brave/components/brave_vpn/browser/brave_vpn_service.h"
+#endif
+
+namespace webcompat_reporter {
+
+namespace {
+
+class WebcompatReporterDOMHandler : public content::WebUIMessageHandler {
+ public:
+  explicit WebcompatReporterDOMHandler(Profile* profile);
+  WebcompatReporterDOMHandler(const WebcompatReporterDOMHandler&) = delete;
+  WebcompatReporterDOMHandler& operator=(const WebcompatReporterDOMHandler&) =
+      delete;
+  ~WebcompatReporterDOMHandler() override;
+
+  // WebUIMessageHandler implementation.
+  void RegisterMessages() override;
+
+ private:
+  void InitAdditionalParameters(Profile* profile);
+  void HandleSubmitReport(const base::Value::List& args);
+
+  std::unique_ptr<webcompat_reporter::WebcompatReportUploader> uploader_;
+  std::string ad_block_list_names_;
+  std::string languages_;
+  bool language_farbling_enabled_ = true;
+  bool brave_vpn_connected_ = false;
+};
+
+WebcompatReporterDOMHandler::WebcompatReporterDOMHandler(Profile* profile)
+    : uploader_(std::make_unique<webcompat_reporter::WebcompatReportUploader>(
+          profile->GetURLLoaderFactory())) {
+  InitAdditionalParameters(profile);
+}
+
+void WebcompatReporterDOMHandler::InitAdditionalParameters(Profile* profile) {
+  std::vector<std::string> ad_block_list_names;
+
+  // Collect all enabled adblock list names
+  brave_shields::AdBlockService* ad_block_service =
+      g_brave_browser_process->ad_block_service();
+  if (ad_block_service != nullptr) {
+    brave_shields::AdBlockRegionalServiceManager* regional_service_manager =
+        ad_block_service->regional_service_manager();
+    CHECK(regional_service_manager);
+    for (const brave_shields::FilterListCatalogEntry& entry :
+         regional_service_manager->GetFilterListCatalog()) {
+      if (regional_service_manager->IsFilterListEnabled(entry.uuid)) {
+        ad_block_list_names.push_back(entry.title);
+      }
+    }
+  }
+
+  ad_block_list_names_ = base::JoinString(ad_block_list_names, ",");
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
```

Comment:
the UI code is not the correct place to put this because it's creating an unnecessary dependency from `//brave/browser/ui` -> `//brave/browser/tor` which already has deps issues and require check_includes = false. This should be encapsulated in another interface for report generation that only exposes a header target to `//brave/browser/ui`